sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -44,6 +50,27 @@ public final class CoreUtils {
 
     private static final char[] LOWERCASE_HEX_CHARACTERS = "0123456789abcdef".toCharArray();
 
+    // The thread pool size for the shared executor service.
+    //
+    // This uses the configuration setting 'azure.sdk.threadPoolSize' if set, otherwise it defaults to 10 times the
+    // number of available processors.
+    // If 'azure.sdk.threadPoolSize' is set to a non-integer, negative value, or zero, the default value is used.
+    private static final int THREAD_POOL_SIZE
+        = Configuration.getGlobalConfiguration().get("azure.sdk.threadPoolSize", config -> {
+        try {
+            int size = Integer.parseInt(config);
+            if (size <= 0) {
+                return 10 * Runtime.getRuntime().availableProcessors();
+            } else {
+                return size;
+            }
+        } catch (NumberFormatException ignored) {
+            return 10 * Runtime.getRuntime().availableProcessors();
+        }
+    });
+
+    private static final AtomicReference<ExecutorService> SHARED_EXECUTOR_SERVICE = new AtomicReference<>();
```

Comment:
I really want us to be conscious of what newer JDKs have around structured concurrency and virtual threads. I think this, especially if you have an actual type rather than this being in CoreUtils, should be part of our MR-JAR story.